### ISSUE TITLE
Does it exist a weights converter between the different API of the DynUNet?

### ISSUE THREAD

**Issue by LucasFidon** [Project-MONAI/MONAI#2158]
I am wondering if it exists a script to convert weights between past and present API of the DynUNet.

## My problem
I have been working with the API of the DynUNet of MONAI 0.4.0+85.gaf1ffd6 for a few months.
Unfortunately, some breaking changes in the API of the DynUNet have happened somewhere between MONAI 0.4.0 and MONAI 0.5.0. As a result, I cannot load the weights of my previous DynUNet in MONAI >=0.5.0.

## My ugly solution
I am now working with a copy of the DynUNet of MONAI 0.4.0+85.gaf1ffd6.
Note that I have found that the DynUNet of MONAI 0.4.0+85.gaf1ffd6 cannot be found in any of the releases of MONAI; it is different from the API in 0.4.0 and 0.5.0rc1.

**Comment by rijobro** [comment]
Hi @LucasFidon, unfortunately as MONAI is in 0.x release still, breaking changes are somewhat unavoidable. We had a meeting last week about how to avoid these in the future to avoid the sort of problems you're seeing.

One solution that might sidestep this problem would be to checkout 0.4.x that's compatible with your code, and then save the model using `torch.script.save`. You could then checkout any other version (e.g., 0.5.x) and load the model back in with `torch.script.load`. Would that work for you or have I overlooked something?

@wyli @Nic-Ma @ericspod I really like @charliebudd 's idea of saving networks and loading them back in as a unit test. That way, we'll know if we break backwards compatibility. We could also save the result of `model.forward()` to check for consistency.

**Comment by Nic-Ma** [comment]
@rijobro , yes, it would be great to enhance the unit tests of our networks, we didn't get a chance to implement more tests for exact forward results before.

Thanks.

**Comment by charliebudd** [comment]
Tangential point, unit testing the outputs of whole networks may pose some problems. I've run into a few issues here #2023, which I'm begining to suspect stems from pytorch's convolution, admittedly only when given very small values. [Pytorch's reproducibility toolkit](https://pytorch.org/docs/stable/notes/randomness.html) has a few new features in 1.8.1 which are flagged as in beta release, but we should certainly look at including these in our unit testing procedure.

**Comment by ericspod** [comment]
We'd like to use more of the new API but can't rely on things that aren't in versions that we support. As a simpler sanity check could we just keep a list of parameter names, types, and dimensions that a network should have and ensure those alone are stored by networks we create to test? This would catch @LucasFidon 's error if he's getting complaint about names not being found when loading weights.

**Comment by charliebudd** [comment]
Agreed, testing the torch.save output is the main take away.
